File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        if (nums.length == 0) return 0;
        Arrays.sort(nums);
        int current = 1;
        int max = current;
        for (int i = 1; i < nums.length; i++) {
            if (nums[i-1] == nums[i] - 1) {
                current++;
                max = Math.max(current, max);
            } else if (nums[i-1] == nums[i]) {
            } else {
                max = Math.max(current, max);
                current = 1;
            }
        }
        return max;
    }
}

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.kt
class Solution {
    fun longestConsecutive(nums: IntArray): Int {
        if (nums.size == 0) return 0
        Arrays.sort(nums);
        var max = 1;
        var current = 1;
        for (i in 0 until nums.size-1) {
            if (nums[i] == nums[i+1]) continue;
            if (nums[i]+1 == nums[i+1]) current++;
            else current = 1;
            max = Math.max(current, max)
        }
        return max;
    }
}